qiskit/transpiler/coupling.py

```diff
@@ -268,9 +276,6 @@ def reduce(self, mapping):
         Raises:
             CouplingError: Reduced coupling map must be connected.
         """
```

Comment:
I can revert this part, for now. But, I viewed it as removing a previous restriction rather than a breaking change because  we're making the class support disjoint coupling maps more broadly. I don't think it's strictly required for this PR though, because we can still restrict the `reduce()` method to no allow returning a disjoint graph.

Reply:
Right, agree, and I don't think we've discussed in depth anywhere whether or not `raises` behavior is a place where users should expect consistency, but certainly it's possible there's code that depends on it, e.g. 
https://github.com/Qiskit/qiskit-terra/blob/eb4c5cbeb5200414a7341c02fa5836e6509c0834/qiskit/transpiler/passes/routing/algorithms/bip_model.py#L62

I think leaving this function as is or updating it are both reasonable, but if we update it, it should probably get a warning and a release note.